share/rocm/cmake/ROCMCreatePackage.cmake

```diff
@@ -274,3 +318,45 @@ macro(rocm_set_comp_cpackvar HEADER_ONLY components)
         set(CPACK_DEBIAN_${COMPONENT}_FILE_NAME "DEB-DEFAULT")
     endforeach()
 endmacro()
+
+macro(rocm_package_setup_component COMPONENT_NAME)
+    set(options)
+    set(oneValueArgs PACKAGE_NAME LIBRARY_NAME PARENT)
+    set(multiValueArgs DEPENDS)
+
+    cmake_parse_arguments(PARSE "${options}" "${oneValueArgs}" "${multiValueArgs}" ${ARGN})
+
+    list(APPEND ROCM_PACKAGE_COMPONENTS ${COMPONENT_NAME})
+
+    if(NOT DEFINED PARSE_PACKAGE_NAME)
+        string(TOLOWER "${COMPONENT_NAME}" PARSE_PACKAGE_NAME)
+        if(NOT BUILD_SHARED_LIBS)
+            set(PARSE_PACKAGE_NAME "${PARSE_PACKAGE_NAME}-static")
+        endif()
+    endif()
+
+    if(NOT DEFINED PARSE_LIBRARY_NAME)
+        set(PARSE_LIBRARY_NAME "${PROJECT_NAME}")
+    endif()
+
+    string(TOUPPER "${COMPONENT_NAME}" COMPONENT_GNAME)
+
+    set(CPACK_DEBIAN_${COMPONENT_GNAME}_PACKAGE_NAME "${PARSE_LIBRARY_NAME}-${PARSE_PACKAGE_NAME}")
+    set(CPACK_RPM_${COMPONENT_GNAME}_PACKAGE_NAME "${PARSE_LIBRARY_NAME}-${PARSE_PACKAGE_NAME}")
+
+    if(DEFINED PARSE_PARENT)
+        list(APPEND ROCM_PACKAGE_COMPONENT_DEPENDENCIES "${PARSE_PARENT}->${COMPONENT_NAME}")
+    endif()
+
+    if(DEFINED PARSE_DEPENDS)
+        cmake_parse_arguments(DEPENDS "RUNTIME" "" "COMMON;DEBIAN;RPM;COMPONENT" ${PARSE_DEPENDS})
+        rocm_package_add_deb_dependencies(COMPONENT ${COMPONENT_NAME} DEPENDS ${DEPENDS_COMMON} ${DEPENDS_DEBIAN})
+        rocm_package_add_rpm_dependencies(COMPONENT ${COMPONENT_NAME} DEPENDS ${DEPENDS_COMMON} ${DEPENDS_RPM})
+        foreach(DEP_COMP IN LISTS DEPENDS_COMPONENT)
+            list(APPEND ROCM_PACKAGE_COMPONENT_DEPENDENCIES "${COMPONENT_NAME}->${DEP_COMP}")
```

Comment:
So why couldn't you do `rocm_package_add_deb_dependencies(COMPONENT ${COMPONENT_NAME} DEPENDS ${CPACK_DEBIAN_${DEP_COMP}_PACKAGE_NAME})` to add the dependencies on other components?

Reply:
The idea of doing it this way was that you can reference a component which hasn't been set up yet and still end up with the correct name, which was primarily to deal with the `Unspecified` (runtime) and `devel` components, which have their names set in `rocm_create_package`. Because this way the component dependencies aren't resolved to package dependencies until the very end of `rocm_create_package`, we can ensure that the dependencies use the same names as the generated packages.

Reply:
We could also use this to ensure the `NAME` argument to `rocm_create_package` is properly propagated to the other components. Currently, unless the `LIBRARY_NAME` parameter is passed to `rocm_package_setup_component`, it just uses `PROJECT_NAME`, which is not always correct (e.g. when generating rocBLAS for Nvidia gpus, the project name is still `rocblas`, but we would like the package name to be `rocblas-alt`). Currently the components would be named with `rocblas`, but we could put in a placeholder when setting up the components, and finalize the names in the call to `rocm_create_package`.